lib/controllers/users.js

```diff
@@ -0,0 +1,178 @@
+ * @param  {object} authUtils an instance of the authentication utilities
+ * @param  {object} userModel bookshelf model to fetch user data
+ * @param  {object} userData  object containing properties to match the user on
+ * @return {object}          undefined if no matches found,
+ *                                 otherwise a bookshelf object
+ */
+internals.getUser = (authUtils, UserModel, userData) => {
+    const user = new UserModel();
+    const formatQuery = user._utils.formatQuery;
+
+    //clone to avoid mutation
+    let username = userData.username;
+    let qbObj = {orWhere: formatQuery(userData)};
+    if (username) {
+        qbObj.orWhere.username = authUtils.encryptUsername(username);
+    }
+
+    return user
+        .query(qbObj)
+        .fetch();
+};
+
+internals.defaultValues = {
+    siteId: '-1',
+    activeFlag: 'Y',
+    passwordExpDate: moment().add(1, 'years').format('MM/DD/YYYY'),
+    acctExpDate: moment().add(1, 'years').format('MM/DD/YYYY')
+};
+
+/**
+ * Call the toJSON method on the object
+ * @param  {Object} obj object to call the toJSON method on (bookshelf)
+ * @return {string}     JSON
+ */
+internals.callToJson = (obj) => {
+    return obj.toJSON();
+};
+
+/**
+ * Call the fetch method on the object
+ * @param  {Object} obj object to call the fetch method on (bookshelf)
+ * @return {Promise}
+ */
+internals.callFetch = (obj) => {
+    //remove all attributes but username
+    obj.attributes = { username: obj.get('username') };
+    return obj.fetch();
+};
+
+/**
+ * hide protected fields so that they are no returned
+ * @param  {object} obj bookshelf object of user
+ * @return {object}     sanitized model
+ */
+internals.sanitize = (obj) => {
+    delete obj.attributes.passwordHash;
+    delete obj.attributes.passwordResetKey;
+    return obj;
+};
+
+/**
+ * Add new user to database
+ * @param  {object} userModel bookshelf model to fetch user data
+ * @param  {object} userData  object containing properties to match the user on
+ * @return {[type]}           [description]
+ */
+internals.addUser = (authUtils, UserModel, userData) => {
+    const userDataClone = _.clone(userData);
+    const username = userDataClone.username;
+    const password = userDataClone.password;
+    const setHashedPassword = (password) => {
+        userDataClone.passwordHash = password;
+        userDataClone.password = null;
+        return userDataClone;
+    };
+
+    const saveUser = (userData) => {
+        return UserModel.forge(userData).save(null, {method: 'insert'});
+    };
+
+    if (username) {
+        userDataClone.username = authUtils.encryptUsername(username);
+    }
+
+    return authUtils.hashPassword(password)
+        .then(setHashedPassword)
+        .then(_.partialRight(_.defaults, internals.defaultValues))
+        .then(saveUser);
+};
+
+module.exports.create = {
+    tags: ['api', 'users'],
+    notes: [
+        'Creates a new user with the properties in the payload.'
+    ].join('<br>'),
```

Comment:
Notes will have a tendency to grow as we increase our logic. Using the array will save time in the future when we need multiple lines. 

Looking at it another way: The array is part of the template in _CONTRIBUTING.md_, and I didn't want to remove it entirely.